rts/lra/client/src/main/java/io/narayana/lra/client/NarayanaLRAClient.java

```diff
@@ -265,13 +287,17 @@ public URI startLRA(URI parentLRA, String clientID, Long timeout, ChronoUnit uni
                 .queryParam(TIMELIMIT_PARAM_NAME, Duration.of(timeout, unit).toMillis())
                 .queryParam(PARENT_LRA_PARAM_NAME, encodedParentLRA)
                 .request()
-                .post(null);
+                .async()
+                .post(null)
+                .get(CLIENT_TIMEOUT, TimeUnit.SECONDS);
 
             // validate the HTTP status code says an LRA resource was created
             if (isUnexpectedResponseStatus(response, Response.Status.CREATED)) {
                 String responseEntity = response.hasEntity() ? response.readEntity(String.class) : "";
-                LRALogger.i18NLogger.error_lraCreationUnexpectedStatus(response.getStatus(), responseEntity);
-                throwGenericLRAException(null, INTERNAL_SERVER_ERROR.getStatusCode(),
+                if (verbose) {
```

Comment:
I wonder what is the benefit for having this `verbose` flag here. My opinion is that when the LRA is not created it's generally usable to understand the reason. The `throwGenericLRAException` was changed to contain this information in case of error. Then the `verbose` flag won't provide more information, will it?

Reply:
The code is generating a stacktrace and that is not useful. The user wants a new LRA but the system responsible for fulfilling that request is not available. I don't think that is a good reason to start generating stack traces in the logs (this is a distributed system and Availability cannot be guaranteed - ie it is an expected condition).

LRA operates in an environment where failure/unavailability conditions are the norm and dumping stacktraces is not the way to go, it gives the sense that the s/w is buggy. We only need to generate a decent response back to the client and a warning in the logs for further analysis and correlation. And this is what this PR is trying to achieve.

Reply:
Exactly right, we should not be throwing exceptions and my code changes log the warning but avoid the exception.
The verbosity flag was added because I don't want the server side filters throwing exceptions. Existing code using this client API is unaffected (I still think it's wrong even for them but it isn't the subject of my PR).